testing-common/src/main/java/io/opentelemetry/javaagent/testing/common/AgentTestingExporterAccess.java

```diff
@@ -0,0 +1,277 @@
+import io.opentelemetry.proto.common.v1.KeyValue;
+import io.opentelemetry.proto.resource.v1.Resource;
+import io.opentelemetry.proto.trace.v1.InstrumentationLibrarySpans;
+import io.opentelemetry.proto.trace.v1.ResourceSpans;
+import io.opentelemetry.proto.trace.v1.Span;
+import io.opentelemetry.proto.trace.v1.Status;
+import io.opentelemetry.sdk.common.InstrumentationLibraryInfo;
+import io.opentelemetry.sdk.testing.trace.TestSpanData;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public final class AgentTestingExporterAccess {
+  private static final char TRACESTATE_KEY_VALUE_DELIMITER = '=';
+  private static final char TRACESTATE_ENTRY_DELIMITER = ',';
+  private static final Pattern TRACESTATE_ENTRY_DELIMITER_SPLIT_PATTERN =
+      Pattern.compile("[ \t]*" + TRACESTATE_ENTRY_DELIMITER + "[ \t]*");
+
+  private static final MethodHandle getExportRequests;
+  private static final MethodHandle reset;
+
+  static {
+    try {
+      Class<?> agentTestingExporterFactoryClass =
+          AgentClassLoaderAccess.loadClass(
+              "io.opentelemetry.javaagent.testing.exporter.AgentTestingExporterFactory");
+      getExportRequests =
+          MethodHandles.lookup()
+              .findStatic(
+                  agentTestingExporterFactoryClass,
+                  "getExportRequests",
+                  MethodType.methodType(List.class));
+      reset =
+          MethodHandles.lookup()
+              .findStatic(
+                  agentTestingExporterFactoryClass, "reset", MethodType.methodType(void.class));
+    } catch (Exception e) {
+      throw new Error("Error accessing fields with reflection.", e);
+    }
+  }
+
+  public static void reset() {
+    try {
+      reset.invokeExact();
+    } catch (Throwable t) {
+      throw new Error("Could not invoke reset", t);
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  public static List<SpanData> getExportedSpans() {
+    final List<byte[]> exportRequests;
+    try {
+      exportRequests = (List<byte[]>) getExportRequests.invokeExact();
+    } catch (Throwable t) {
+      throw new Error("Could not invoke getExportRequests", t);
+    }
+
+    List<ResourceSpans> allResourceSpans =
+        exportRequests.stream()
+            .map(
+                serialized -> {
+                  try {
+                    return ExportTraceServiceRequest.parseFrom(serialized);
+                  } catch (InvalidProtocolBufferException e) {
+                    throw new Error(e);
+                  }
+                })
+            .flatMap(request -> request.getResourceSpansList().stream())
+            .collect(Collectors.toList());
+    List<SpanData> spans = new ArrayList<>();
+    for (ResourceSpans resourceSpans : allResourceSpans) {
```

Comment:
Annoying - but seems a simple way to get exported spans across the classloader. However, I will play with whether Jackson can marshal them for free (the `SpanWrapper` worries me)